snomed/com.b2international.snowowl.validation.snomed/src/main/resources/scripts/rule664.groovy

```diff
@@ -89,13 +89,11 @@ if (params.isUnpublishedOnly) {
 		)
 		.limit(10_000)
 		.build())
-		.each { Hits<String[]> publishedTermsBatch ->
-			// all returned terms are duplicate of an unpublished term
-			publishedTermsBatch.each { publishedTerm ->
+		.each { publishedTerm ->
```

Comment:
We have discussed this in person, but I still can't see how this would work. Groovy's extension method simply calls `iterator()` on the `Stream` using reflection (it has no dedicated version for handling instances of `Stream`):

```groovy
/**
 * Iterates through an aggregate type or data structure,
 * passing each item to the given closure.  Custom types may utilize this
 * method by simply providing an "iterator()" method.  The items returned
 * from the resulting iterator will be passed to the closure.
 *
 * @param self    the object over which we iterate
 * @param closure the closure applied on each element found
 * @return the self Object
 * @since 1.0
 */
public static <T> T each(T self, Closure closure) {
  each(InvokerHelper.asIterator(self), closure);
  return self;
}
```

There is no `flatMap`-like behavior that would unpack instances of `Hits<String[]>` into individual `String[]` elements.

I have also added `println`s in the closure and the output was as follows:

```
[2023-07-28T08:16:21.059] INFO  Worker-0: validationJob                          validation                                                       Executing rule '664'...
Query starts
Published term is Hits{hits=[[Ljava.lang.String;@6f0e9267], limit=10000, total=1, searchAfter=AoEpNjY0MjI0MDE0}
publishedTerm[1] is null
publishedTerm[2] is null
Query finished
```